lib/ecto/type.ex

```diff
@@ -1160,6 +1113,14 @@ defmodule Ecto.Type do
     end
   end
 
+  defp equal_fun({:parameterized, mod, opts}) when is_atom(mod) do
+    if loaded_and_exported?(mod, :equal?, 3) do
+      fn a, b ->
+        mod.equal?(a, b, opts)
+      end
+    end
+  end
+
```

Comment:
As we are not removing any `{:embed, _}` match is it one necessary?

Reply:
Embeds didn't previously pass through the cast function, so they never hit [this code path](https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/changeset.ex#L561). Now that they do, we need to be able to test the equality of the parameterized type after casting.